### ISSUE TITLE
Doesn't detect some links when prefixed with colon or equals

### ISSUE THREAD

**Issue by xPaw** [markdown-it/linkify-it#62]
Two examples:
```
test:www.google.com
test=https://www.google.com
```

While opposite cases get detected:
```
test=www.google.com
test:https://www.google.com
```

However it detects `test=www.google.com` as the entire link.

**Comment by puzrin** [comment]
IMHO all those examples have broken punctuation and don't need fix. Where to you have such input in real world?

**Comment by xPaw** [comment]
I'm dealing with user input in a chat app, so the input is unpredictable. I created this issue because the other parser we're currently using allows for a punctuation before www.

https://medialize.github.io/URI.js/docs.html#static-withinString is where it works.

**Comment by xPaw** [comment]
Another two "interesting" cases that are different (from our test suite) but I'm not sure which ones are correct:

Matched by URI.js by not linkify-it:
```
ssh://-oProxyCommand=whois
```

Matched by linkify-it but not URIjs:
```
http://a:%p@c
```

**Comment by puzrin** [comment]
You put too many different cases into one issue. If `ssh` proto support required, you need to extend existing schemas (see readme and dig src).

Since text structure is not determined, all such parsers are euristic. That means a reasonable tradeoff between sensitivity and false positives. If some parser better understand some cases, that doesn't mean it's better, because in may have more false positives. For example, i would not recomment to enable fuzzy links (without schemas), because of nasty side effects possible.

Could you summarize desired improvement (and think again, if it's really needed so much)? Allow `=` before known schema name, without space?

**Comment by xPaw** [comment]
`ssh` is enabled schema, it also happens with `http:`. Sorry for putting too many issues here, I didn't want to create different issues as that might be more annoying.

> Allow = before known schema name, without space?

Sure, because other characters as far as I can see get detected just fine (e.g. @ & : . !), but `=` or `~` do not.

**Comment by puzrin** [comment]
Is `~` really necessary? We should add only minimal possible set.

- `test=https://www.google.com` - this can be fixed, i think
- `test:www.google.com` - this should not be fixed, IMHO (possible side effects).

**Comment by xPaw** [comment]
> Is ~ really necessary?

It's not, I'm just providing an example. Is there a particular reason there's a set list of characters that are allowed before a protocol?

**Comment by puzrin** [comment]
> Is there a particular reason there's a set list of characters that are allowed before a protocol?

All those have resonable examples in real world. Don't remember details. Good start point is to check current test cases.

**Comment by puzrin** [comment]
> ssh is enabled schema

If you implemented it as alias to `http`, that may be not correct, because `http` requires correct domain name to follow. In your example `ssh` is followed not by domain name.

**Comment by xPaw** [comment]
> In your example ssh is followed not by domain name.

That's actually correct, sorry about that!

What about the `http://a:%p@c` example? In that case percent encoding is incorrect.

**Comment by puzrin** [comment]
Looks like username+pass for auth. I dont' remember details, see rexexps. Anyway, it's wrong approach to try all possible samples. Detector use another principle - it guess only USEABLE cases, embedded into ordinary text.

**Comment by xPaw** [comment]
> Anyway, it's wrong approach to try all possible samples.

I fully understand, I'm not trying to get every possible case fixed. Just making sure that these cases were accounted for already, so that we don't run into the same issues we did run into before.

**Comment by puzrin** [comment]
I think, to make things more constructive, it's better to create a new issue with list of your bad cases with comments (examples) how those happened, frequencies and importance (your opinion). Then i'll be able to reply all at once. Current issue tend to become "infinite" and that's a bit inconvenient.

For now, i recorded that `=` could be allowed before protocol, when space is omitted due mistype.

**Comment by puzrin** [comment]
I mean, every issue should have some resolution, when we can close it. If we add more and more conditions, issue become endless. I do not reject to fix problems, when possible, but we should classify those more effectively.

**Comment by puzrin** [comment]
Let's close this. Please, create separate issues for each kind of problem, if still actual.

Default answer is "don't try to fix mistypes", but i will try to check if something possible without side effects.
